### PR TITLE
uiSenderFixed now looks up the given account name in the wallet

### PR DESCRIPTION
**Pull request by mankyKitty**

### CODE REVIEW

File: frontend/src/Frontend/Wallet.hs [hunk 1]
```diff
@@ -140,6 +141,21 @@ findFirstVanityAccount predicate as = do
...
+  -> ChainId
+  -> AccountName
+  -> Dynamic t (Maybe (Some AccountRef))
+findAccountByName w netname cid n = ffor (_wallet_accounts w) $ \accStorage ->
+  Map.lookup netname (unAccountStorage accStorage) >>=
+    fmap getFirst . getOption . foldAccounts g
+  where
+    g acc@(aref :=> _) = if accountToName acc == n && accountChain acc == cid
+      then Option . Just . First $ Some aref
+      else Option Nothing
```
Comment (alexfmpe) [inline frontend/src/Frontend/Wallet.hs]: [Option](https://hackage.haskell.org/package/base-4.12.0.0/docs/Data-Semigroup.html#t:Option)'s getting deprecated in ghc 8.8 and removed in 8.10. I think `Alt Maybe` would work here

File: frontend/src/Frontend/UI/DeploymentSettings.hs [hunk 2]
```diff
@@ -775,8 +794,8 @@ uiSenderDropdown uCfg m chainId setSender = do
           (Map.singleton Nothing)
           (Map.insert Nothing "Choose an Account" . Map.mapKeys Just)
       <$> textAccounts
-  choice <- dropdown Nothing dropdownItems $ uCfg
-    & dropdownConfig_setValue .~ leftmost [Nothing <$ updated chainId, setSender]
+  choice <- dropdown Nothing (traceDyn "dropdownItems" dropdownItems) $ uCfg
```
Comment (alexfmpe) [inline frontend/src/Frontend/UI/DeploymentSettings.hs]: leftover `traceDyn`

File: frontend/src/Frontend/UI/DeploymentSettings.hs [hunk 3]
```diff
@@ -775,8 +794,8 @@ uiSenderDropdown uCfg m chainId setSender = do
...
+    & dropdownConfig_setValue .~ leftmost [Nothing <$ updated chainId, traceEvent "setSender" setSender]
```
Comment (alexfmpe) [inline frontend/src/Frontend/UI/DeploymentSettings.hs]: lefover `traceEvent`

File: frontend/src/Frontend/UI/DeploymentSettings.hs [hunk 4]
```diff
@@ -727,12 +727,31 @@ uiMetaData m mTTL mGasLimit = do
...
+        c <- MaybeT chainId
+        MaybeT $ findAccountByName (model ^. wallet) n c sender
+
   _ <- uiInputElement $ def
     & initialAttributes %~ Map.insert "disabled" ""
     & inputElementConfig_initialValue .~ unAccountName sender
-  pure $ pure $ pure $ Some $ AccountRef_Vanity sender "chain" -- TODO check the chain
+    & inputElementConfig_setValue .~ ffor (updated mAccRef) (\case
+        Nothing -> "Account name not found!"
+        Just (Some a) -> accountRefToName a
```
Comment (alexfmpe) [inline frontend/src/Frontend/UI/DeploymentSettings.hs]: Is `accountRefToName a` ever different than `sender` or are just checking existance?

Comment (mankyKitty) [reply frontend/src/Frontend/UI/DeploymentSettings.hs]: By that stage there should be no difference.

**Pull request merged**